sale_loyalty_order_line_link/migrations/16.0.1.0.0/pre-migration.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2023 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+_renamed_fields = [
+    (
+        "sale.order.line",
+        "sale_order_line",
+        "coupon_program_id",
+        "loyalty_program_id",
+    ),
+]
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.rename_fields(env, _renamed_fields)
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE sale_order_line
+        ADD COLUMN IF NOT EXISTS reward_id INTEGER;
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE sale_order_line AS sol
+        SET sol.reward_id = lr.id
+        FROM loyalty_reward AS lr
+        WHERE sol.loyalty_program_id = lr.program_id
+        AND sol.loyalty_program_id IS NOT NULL;
```

Comment:
No need of this condition, as you are already matching with loyalty_reward and only rows with `loyalty_program_id` filled will match.